packages/console/src/components/Button/index.tsx

```diff
@@ -32,22 +35,56 @@ const Button = ({
   title,
   icon,
   className,
+  isLoading = false,
+  loadingDelay = 500,
+  onClick,
   ...rest
 }: Props) => {
   const { t } = useTranslation();
+  const [showSpinner, setShowSpinner] = useState(false);
+  const timeout = useRef<number>();
+
+  useEffect(() => {
+    // Delay showing the spinner after 'loadingDelay' milliseconds
+    if (isLoading) {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      timeout.current = setTimeout(() => {
+        setShowSpinner(true);
+      }, loadingDelay);
+    } else {
+      clearTimeout(timeout.current);
+      setShowSpinner(false);
+    }
+
+    return () => {
+      clearTimeout(timeout.current);
+    };
+  }, [isLoading, loadingDelay]);
 
   return (
     <button
       className={classNames(
         styles.button,
         styles[type],
         styles[size],
-        conditionalString(icon && styles.withIcon),
+        icon && styles.withIcon,
+        isLoading && styles.loading,
         className
       )}
       type={htmlType}
+      onClick={(event) => {
+        if (isLoading) {
+          return false;
```

Comment:
In this state, the button is "disabled" in code logic, but is "clickable" in UI, not so properly?

Reply:
this state means isLoading but not showing spinner.

Reply:
No, `isLoading` has a `user-select: none;` effect on the button. So they are consistent